In [1]:
import matplotlib.pyplot as plt
import pandas as pd

In [2]:
titles = pd.read_csv('titles.csv')
casts = pd.read_csv('cast.csv')
release = pd.read_csv('release_dates.csv')

In [44]:
titles.head()

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

In [45]:
casts.head()

title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4  31.0
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

In [46]:
release.head()

title  year      country        date
0   #73, Shaanthi Nivaasa  2007        India  2007-06-15
1                   #BKKY  2016     Cambodia  2017-10-12
2                 #Beings  2015      Romania  2015-01-29
3               #Captured  2017          USA  2017-09-05
4  #Ewankosau saranghaeyo  2015  Philippines  2015-01-21

### How many movies are listed in the titles dataframe?

In [4]:
titles.title.nunique()

223828

**Titles have 223828 movies listed in it**

### What are the earliest two films listed in the titles dataframe?

In [5]:
titles.sort_values('year').head(2)

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

The earliest two films are ***Miss Jerry(1894)*** followed by ***The Startled Lover(1898)*** 

### How many movies have the title "Hamlet"?

In [9]:
print(f"The are {titles[titles['title']=='Hamlet'].shape[0]} movies with the title Hamlet")

The are 20 movies with the title Hamlet


### How many movies are titled "North by Northwest"?

In [10]:
print(f"The are {titles[titles['title']=='North by Northwest'].shape[0]} movies with the title 'North by Northwest'")

The are 1 movies with the title 'North by Northwest'


### When was the first movie titled "Hamlet" made?

In [30]:
hamelet = titles[titles['title']=='Hamlet'].sort_values('year').head(1).year
hamelet

52521    1910
Name: year, dtype: int64

The first movie titled Hamlet was made on ***1910***

### List all of the "Treasure Island" movies from earliest to most recent.

In [32]:
treasIsland = titles[titles['title']=='Treasure Island'].sort_values('year') 
treasIsland

title  year
55686   Treasure Island  1918
196514  Treasure Island  1920
141822  Treasure Island  1934
77354   Treasure Island  1950
65051   Treasure Island  1972
37238   Treasure Island  1973
173051  Treasure Island  1985
27076   Treasure Island  1999

### How many movies were made in the year 1950?

In [34]:
print(f"{titles[titles.year == 1950].shape[0]} movies were made in the year 1950")

1113 movies were made in the year 1950


### How many movies were made in the year 1960?

In [35]:
print(f"{titles[titles.year == 1960].shape[0]} movies were made in the year 1950")

1549 movies were made in the year 1950


### How many movies were made from 1950 through 1959?

In [36]:
print(f"{titles[(titles.year>=1950) & (titles.year<1959)].shape[0]} movies were made through the years 1950 to 1959")

11419 movies were made through the years 1950 to 1959


### In what years has a movie titled "Batman" been released?

In [39]:
titles[titles.title=='Batman'].year

39349     1943
240896    1989
Name: year, dtype: int64

Batman was released in the years **1943 and 1989**

### How many roles were there in the movie "Inception"?

In [43]:
casts[casts['title']=='Inception'].character.shape[0]

78

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [47]:
casts[casts['title']=='Inception'].isnull().sum().sum()

27

### But how many roles in the movie "Inception" did receive an "n" value?

In [48]:
casts[casts['title']=='Inception'].shape[0] - casts[casts['title']=='Inception'].isnull().sum().sum()

51

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [50]:
temp = casts[(casts['title']=='North by Northwest') & (casts['n'].notnull())]

In [51]:
temp[['name','n']].sort_values('n')

name     n
871426             Cary Grant   1.0
3518333       Eva Marie Saint   2.0
1457165           James Mason   3.0
3157232   Jessie Royce Landis   4.0
356707         Leo G. Carroll   5.0
3048272  Josephine Hutchinson   6.0
1695173           Philip Ober   7.0
1275337         Martin Landau   8.0
2447081         Adam Williams   9.0
1811387          Edward Platt  10.0
665768      Robert Ellenstein  11.0
2295714          Les Tremayne  12.0
464736        Philip Coolidge  13.0
1508344         Patrick McVey  14.0
206471           Edward Binns  15.0
1384780             Ken Lynch  16.0

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [52]:
temp = casts[(casts['title']=='Sleuth') & (casts['n'].notnull()) & (casts['year']==1972)]

In [53]:
temp[['name','n']].sort_values('n')

name    n
1705394    Laurence Olivier  1.0
326773        Michael Caine  2.0
373876       Alec Cawthorne  3.0
1465483  John (II) Matthews  4.0
2724553  Eve (III) Channing  5.0
1448690        Teddy Martin  6.0

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [54]:
temp = casts[(casts['title']=='Sleuth') & (casts['n'].notnull()) & (casts['year']==2007)]


In [55]:
temp[['name','n']].sort_values('n')

name    n
326774   Michael Caine  1.0
1293634       Jude Law  2.0
1805907  Harold Pinter  3.0

### How many roles were credited in the silent 1921 version of Hamlet?

In [56]:
casts[(casts['title']=='Hamlet') & (casts['year']==1921)].character.shape[0]


9

### How many roles were credited in Branagh’s 1996 Hamlet?

In [57]:
casts[(casts['title']=='Hamlet') & (casts['year']==1996)].character.shape[0]


54

### How many "Hamlet" roles have been listed in all film credits through history?

In [58]:
casts[casts['title']=='Hamlet'].character.shape[0]

342

### How many people have played an "Ophelia"?

In [ ]:
casts[casts['title']==]

### How many people have played a role called "The Dude"?

In [3]:
casts[casts['character']=='The Dude'].character.shape[0]

19

### How many people have played a role called "The Stranger"?

In [4]:
casts[casts['character']=='The Stranger'].character.shape[0]


212

### How many roles has Sidney Poitier played throughout his career?

In [5]:
casts[casts['name']=='Sidney Poitier'].shape[0]


43

### How many roles has Judi Dench played?

In [6]:
casts[casts['name']=='Judi Dench'].shape[0]

55

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [7]:

casts[(casts['name']=='Cary Grant') & (casts['year']==1940) & (casts['n']==2)]

title  year        name   type   character    n
871423  My Favorite Wife  1940  Cary Grant  actor  Nick Arden  2.0

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [8]:
casts[(casts['name']=='Cary Grant') & (casts['year']==1940) & (casts['n']==1)].character

871407          Walter Burns
871448           Matt Howard
871450    C. K. Dexter Haven
Name: character, dtype: object

### How many roles were available for actors in the 1950s?

In [9]:
casts[(casts['year']==1950) & (casts['type']=='actor')].character.shape[0]

16760

### How many roles were available for actresses in the 1950s?

In [10]:

casts[(casts['year']==1950) & (casts['type']=='actress')].character.shape[0]

5637

### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [11]:
casts[(casts['year']<1980) & (casts['n']==1)].shape[0]

63957

### How many non-leading roles were available through from the beginning of film history through 1980?

In [12]:
casts[(casts['year']<1980) & (casts['n']!=1)].shape[0]

1095378

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [13]:
casts[(casts['year']<=1980) & (casts['n'].isnull())].shape[0]

448347